gradle-jdks-setup/src/main/java/com/palantir/gradle/jdks/setup/GradleJdkInstallationSetup.java

```diff
@@ -121,16 +123,39 @@ private static boolean copy(ILogger logger, Path destinationJdkInstallationDirec
                 lockFile, StandardOpenOption.READ, StandardOpenOption.CREATE, StandardOpenOption.WRITE)) {
             channel.lock();
             // double-check, now that we hold the lock
-            if (Files.exists(destinationJdkInstallationDirectory)) {
+            if (Files.exists(destinationJdkInstallationDirectory)
+                    && Files.exists(destinationJdkInstallationDirectory.resolve("bin/java"))) {
                 logger.log(String.format("Distribution URL %s already exists", destinationJdkInstallationDirectory));
                 return false;
             }
             logger.log(
                     String.format("Copying JDK from %s into %s", currentJavaHome, destinationJdkInstallationDirectory));
-            FileUtils.copyDirectory(currentJavaHome, destinationJdkInstallationDirectory);
+            Path tempDirWithPrefix = Files.createTempDirectory("gradle-jdks");
+            FileUtils.copyDirectory(currentJavaHome, tempDirWithPrefix);
+            try {
+                Files.move(
+                        tempDirWithPrefix,
+                        destinationJdkInstallationDirectory,
+                        StandardCopyOption.REPLACE_EXISTING,
+                        StandardCopyOption.ATOMIC_MOVE);
+            } catch (AtomicMoveNotSupportedException e) {
+                logger.log("Cannot move the directory to the final directory, attempting to do a non-atomic operation");
+                Files.move(tempDirWithPrefix, destinationJdkInstallationDirectory, StandardCopyOption.REPLACE_EXISTING);
+            }
+            createSentinelFile(destinationJdkInstallationDirectory);
```

Comment:
What is this sentinel file used for?

Reply:
I don't think it's used elsewhere in this diff and now we have the atomic move the directory existing should be enough.

Reply:
I am thinking that the sentinel file is the flag that tells us if the installation finished successfully. I was thinking that in the future (if we add the check now , we would re-install jdks that are already correctly installed on user's laptops) we might assume that the installation was successful if the file was created.

Reply:
I would hope that since we move the JDK now it will definitely be fine as the move will mean the directory is either there and correct or not there, rather than just partially copied.

Reply:
ack, I'll remove this then